File: src/main/java/com/nao4j/subtrol/service/SubscriptionService.java
package com.nao4j.subtrol.service;

import com.nao4j.subtrol.document.internal.Subscription;
import org.springframework.security.access.annotation.Secured;

import java.util.List;

import static com.nao4j.subtrol.security.Roles.ADMIN;
import static com.nao4j.subtrol.security.Roles.USER;

public interface SubscriptionService {

    @Secured({USER, ADMIN})
    List<Subscription> getAll(String userId, String serviceName);

    @Secured({USER, ADMIN})
    Subscription add(String userId, String serviceName, Subscription subscription);

    @Secured({USER, ADMIN})
    Subscription remove(String userId, String serviceName, Subscription subscription);

}

File: src/main/kotlin/com/nao4j/subtrol/service/SubscriptionService.kt
package com.nao4j.subtrol.service

import com.nao4j.subtrol.model.Subscription
import com.nao4j.subtrol.security.ADMIN
import com.nao4j.subtrol.security.USER
import org.springframework.security.access.annotation.Secured

interface SubscriptionService {

    @Secured(value = [USER, ADMIN])
    fun getAll(userId: String, serviceName: String): List<Subscription>

    @Secured(value = [USER, ADMIN])
    fun add(userId: String, serviceName: String, subscription: Subscription): Subscription

    @Secured(value = [USER, ADMIN])
    fun remove(userId: String, serviceName: String, subscription: Subscription): Subscription

}
